### PR TITLE
First branch

### PR DESCRIPTION
**Pull request by RoyVardi4**

### CODE REVIEW

File: package-lock.json
Comment (reuthaselkorn) [inline package-lock.json:1]: add to .gitignore

File: src/api/Recipes.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+
+class RecipesAPI {
+  static getComplexQueryRecipes = async (queryString: string) => {
+    return axios
+      .get("http://localhost:3000/api/recipes/complexSearch", {
```
Comment (reuthaselkorn) [inline src/api/Recipes.ts]: should we hard code the routes or use variables? 
especially if we have multiple environments

File: src/components/RecipesList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const RecipesList: FC<IList> = ({ queryString }) => {
+  const { isLoading, error, data } = useQuery<{ results: IRecipe[] }, Error>(
+    ["RecipesListComplexSearch", queryString],
+    () => RecipesAPI.getComplexQueryRecipes(queryString)
+  );
+
+  if (isLoading)
+    return (
+      <Grid container spacing={3}>
+        {[1, 2, 3, 4].map((i) => (
```
Comment (reuthaselkorn) [inline src/components/RecipesList/index.tsx:22]: what is this used for? does it stay the same size FOREVER????

File: src/pages/Home/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const Home = () => {
+  const [searchText, setSearchText] = useState("");
+  const debouncedSearchText = useDebounce<string>(searchText, 500)
+
+  return (
+    <Grid container justifyContent="center" rowGap={5}>
+      <Grid item xs={10}>
+        <Typography variant="h6" textAlign="center">
+          Welcome to Reut and Roy's Recipe webapp
```
Comment (reuthaselkorn) [inline src/pages/Home/index.tsx:14]: :)

File: package-lock.json
Comment (RoyVardi4) [reply package-lock.json:1]: Actually package-lock is very important to be committed for Dependency Version Consistency.

Here a article which explains it nicely [click me!](https://medium.com/@InspireTech/should-i-add-package-lock-json-to-git-why-is-it-important-to-commit-package-lock-json-into-git-e8273a8aad00#:~:text=reliable%20software%20deployments.-,Committing%20the%20package%2Dlock.,dependencies%20used%20in%20a%20project.)

File: src/api/Recipes.ts (hunk 1 above)
Comment (RoyVardi4) [reply src/api/Recipes.ts]: I agree, will be done by tomorrow

File: src/components/RecipesList/index.tsx (hunk 2 above)
Comment (RoyVardi4) [reply src/components/RecipesList/index.tsx:22]: Yes it's stay the same size because it is a skeleton to show the user while loading the data

**Pull request merged**